api_tests/registrations/views/test_registration_detail.py

```diff
@@ -442,6 +454,109 @@ def test_read_write_contributor_cannot_update_custom_citation(
         assert res.status_code == 403
 
 
+@pytest.mark.django_db
+class TestRegistrationWithdrawal(TestRegistrationUpdateTestCase):
+
+    @pytest.fixture
+    def public_payload(self, public_registration, make_payload):
+        return make_payload(
+            id=public_registration._id,
+            attributes={'withdrawn': True, 'withdrawal_justification': 'Not enough oopmh.'}
+        )
+
+    def test_withdraw_registration_fails(
+            self, app, user, read_write_contributor, public_registration, make_payload,
+            private_registration, public_url, private_url, public_project, public_payload):
+        # test withdrawal with no auth
+        res = app.put_json_api(public_url, public_payload, expect_errors=True)
+        assert res.status_code == 401
+
+        # test withdrawal from a read write contrib
+        res = app.put_json_api(public_url, public_payload, auth=read_write_contributor.auth, expect_errors=True)
+        assert res.status_code == 403
+
+        # test withdrawal private registration fails
+        payload_private = make_payload(
+            id=private_registration._id,
+            attributes={'withdrawn': True, 'withdrawal_justification': 'fine whatever'}
+        )
+        res = app.put_json_api(private_url, payload_private, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+
+        # test withdrawal component fails
+        project = ProjectFactory(is_public=True, creator=user)
+        NodeFactory(is_public=True, creator=user, parent=project)
+        registration_with_comp = RegistrationFactory(is_public=True, project=project)
+        registration_comp = registration_with_comp._nodes.first()
+        payload_component = make_payload(
+            id=registration_comp._id,
+            attributes={'withdrawn': True}
+        )
+        url = '/{}registrations/{}/'.format(API_BASE, registration_comp._id)
+        res = app.put_json_api(url, payload_component, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
```

Comment:
thanks for this test 👍 - wasn't clear to me that this was being prevented, but I see now that it's enforced model-level